x509/golang.go

```diff
@@ -136,6 +136,42 @@ func SignEstCsr(csrPem string) string {
 	return genTlsCert(csr.Subject, csr.DNSNames, csr.PublicKey)
 }
 
+var oidExtensionReasonCode = []int{2, 5, 29, 21}
+
+func CreateCrl(serials map[string]int) string {
+	factoryKey := factoryCaKeyStorage.loadKey()
+	factoryCa := LoadCertFromFile(FactoryCaCertFile)
+	now := time.Now()
+	crl := &x509.RevocationList{
+		Number:     big.NewInt(1),
+		ThisUpdate: now,
+		NextUpdate: now.Add(time.Minute * 15),
+	}
+	for serial, reason := range serials {
+		num := new(big.Int)
+		if _, ok := num.SetString(serial, 10); !ok {
+			// We expect a valid input here
+			panic("Value is not a valid base 10 serial:" + serial)
```

Comment:
Maybe `DieNotNil(error.New())`?

Reply:
We do validation at the command level.
If we get an invalid serial here - it is a programming error; so I'd like the user to report the full traceback.

Reply:
I thought a user can specify `serials` via the command line. I guess the user specified serial is already verified when this code is executed.

Reply:
Yes, exactly.